src/mon/OSDMonitor.cc

```diff
@@ -14046,6 +14046,16 @@ bool OSDMonitor::prepare_pool_op(MonOpRequestRef op)
 
   const pg_pool_t *pool = osdmap.get_pg_pool(m->pool);
 
+  if (m->op == POOL_OP_CREATE_SNAP ||
```

Comment:
thinking about it more ... since pool level snap IDs are assigned by mons and FS level snap IDs are assigned by the MDS itself, I wonder how the snap IDs may clash causing problems ... which was the main driving factor for this PR
since MDS is never going to talk to the mon for manging FS snaps ... I think I may be lacking a bit of context

@gregsfortytwo could you elaborate on this please

Reply:
The snapids are a shared resource and when you delete a snapshot, you insert it into the osdmap. Any overlap is bad since it would delete both...

Reply:
@gregsfortytwo could you point me to the piece of code on the mds side which initiates the propagation of the deleted snapid to the osdmap

Reply:
@mchangir You'd need to look at SnapServer.cc